app/src/androidTest/java/net/skyscanner/backpack/compose/calendar/BpkCalendarTest.kt

```diff
@@ -0,0 +1,299 @@
+import net.skyscanner.backpack.calendar2.BpkCalendarTestCases
+import net.skyscanner.backpack.calendar2.CalendarParams
+import net.skyscanner.backpack.compose.calendar.internal.CALENDAR_GRID_TEST_TAG
+import net.skyscanner.backpack.demo.compose.BackpackPreview
+import net.skyscanner.backpack.util.InternalBackpackApi
+import org.junit.Assume
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(InternalBackpackApi::class)
+@RunWith(AndroidJUnit4::class)
+class BpkCalendarTest : BpkSnapshotTest() {
+
+  @get:Rule
+  val rule = activityScenarioRule<AppCompatActivity>()
+
+  @get:Rule
+  val composeTestRule = createEmptyComposeRule()
+
+  @Before
+  fun setup() {
+    setDimensions(700, 400)
+    AndroidThreeTen.init(testContext)
+  }
+
+  @Test
+  fun screenshotTestCalendarDefault() {
+    val controller = createController(BpkCalendarTestCases.Params.Default)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestColoredCalendarDefault() {
+    val controller = createController(BpkCalendarTestCases.Params.Colored)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestLabeledCalendarDefault() {
+    val controller = createController(BpkCalendarTestCases.Params.Labeled)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarPast() {
+    val controller = createController(BpkCalendarTestCases.Params.Past)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarLeapFebruary() {
+    val controller = createController(BpkCalendarTestCases.Params.LeapFebruary)
+    controller.setSelection(BpkCalendarTestCases.Selection.LeapFebruary)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarNonLeapFebruary() {
+    val controller = createController(BpkCalendarTestCases.Params.NonLeapFebruary)
+    controller.setSelection(BpkCalendarTestCases.Selection.NonLeapFebruary)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarTodayIsLastDayOfMonth() {
+    val controller = createController(BpkCalendarTestCases.Params.TodayIsLastDayOfMonth)
+    controller.setSelection(BpkCalendarTestCases.Selection.TodayIsLastDayOfMonth)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarTodayIsNewYear() {
+    val controller = createController(BpkCalendarTestCases.Params.TodayIsNewYear)
+    controller.setSelection(BpkCalendarTestCases.Selection.TodayIsNewYear)
+    snap(controller)
+  }
+
+  @Test
+  fun screenshotTestCalendarWithStartDateSelected() {
+    val controller = createController(BpkCalendarTestCases.Params.WithStartDateSelected)
+
+    snap(controller) {
+      it.onAllNodesWithText("17")
+        .onFirst()
+        .performClick()
+        .assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun screenshotTestCalendarWithSameStartAndEndDateSelected() {
+    val controller = createController(BpkCalendarTestCases.Params.WithSameStartAndEndDateSelected)
+
+    snap(controller) {
+
+      it.onAllNodesWithText("17")
+        .onFirst()
```

Comment:
onFirst=Jan, onLast=Fab. No methods in Compose to click on item with index like in View